frontends/web/src/components/transactions/components/details.tsx

```diff
@@ -0,0 +1,187 @@
+import { useEffect, useRef } from 'react';
+import { useTranslation } from 'react-i18next';
+import { ITransaction, IAmount, getTransaction } from '@/api/account';
+import { A } from '@/components/anchor/anchor';
+import { Dialog } from '@/components/dialog/dialog';
+import { FiatConversion } from '@/components/rates/rates';
+import { Amount } from '@/components/amount/amount';
+import { Note } from '@/components/transactions/note';
+import { TxDetail } from './detail';
+import { Arrow } from './arrow';
+import { TxDateDetail } from './date';
+import { TxStatusDetail } from './status';
+import { TxAddressOrTxID } from './address-or-txid';
+import parentStyle from '@/components/transactions/transaction.module.css';
+
+type TProps = {
+  open: boolean;
+  onClose: () => void;
+  accountCode: string;
+  internalID: string;
+  note: string;
+  status: ITransaction['status'];
+  type: ITransaction['type'];
+  numConfirmations: number;
+  numConfirmationsComplete: number;
+  time: string | null;
+  amount: IAmount;
+  sign: string;
+  typeClassName: string;
+  feeRatePerKb: IAmount;
+  explorerURL: string;
+}
+
+export const TxDetailsDialog = ({
+  open,
+  onClose,
+  accountCode,
+  internalID,
+  note,
+  status,
+  type,
+  numConfirmations,
+  numConfirmationsComplete,
+  time,
+  amount,
+  sign,
+  typeClassName,
+  feeRatePerKb,
+  explorerURL,
+}: TProps) => {
+  const { t } = useTranslation();
+
+  const transactionInfo = useRef<ITransaction | null>();
+
+  useEffect(() => {
```

Comment:
I just noticed this causes **all** single transactions to be fetched at once instead only when user clicks on the details. I have to add `if(!transactionInfo.current && open)` to prevent this, will do this later.

Reply:
looks like this was done in last commit, and fixes and issue that was introduced in 2nd last commit?

suqash both commits?